### DISCUSSION TITLE
Error of being not a registered object and segmentation fault

### DISCUSSION THREAD

**Discussion by xchengood** [idaholab/moose#18602] [Q&A Modules: Solid mechanics] [answered]
Hi Moose expert or user,

I am studying and trying to implement a input file example called 3d.i located in the path projects/moose/modules/tensor_mechanics/test/tests/rom_stress_update.

First, I try implementation in terminal and I got the following error.
<img width="709" alt="Screen Shot 2021-08-13 at 10 12 33 AM" src="https://user-images.githubusercontent.com/74226455/129381365-fea75a19-e48f-4d30-bf8f-c0796372fbcf.png">

Then, I try it in the peacock. The following error comes and a window pops out.
<img width="725" alt="Screen Shot 2021-08-13 at 10 13 46 AM" src="https://user-images.githubusercontent.com/74226455/129381656-8c508e5c-c26d-4924-b83d-3cdf9a5b3235.png">

Before implementation, I already update my conda and do something about compile using 'make -j4'.

Could anyone help me with fixing this error? Thank you.

**Comment by dschwen** [comment, marked as answer]
Run the executable with the `--allow-test-objects` option! The SS316 model is just a test for the ROM system.

**Reply by xchengood** [reply]
> --allow-test-objects

Thank you. It works in the terminal. But how can I implement it in peacock? It seems not right if I just put that at the end of the executable command line for peacock. I got 'unrecognized arguments'.

**Reply by dschwen** [reply]
I suggest using an editor like Atom with a MOOSE syntax autocompletion plugin to edit input files, and run them from the terminal. It's a more powerful way to work with MOOSE. Unfortunately peacock is not working properly on the latest macOS for me, so I cannot reproduce your issue.

**Reply by xchengood** [reply]
> I suggest using an editor like Atom with a MOOSE syntax autocompletion plugin to edit input files, and run them from the terminal. It's a more powerful way to work with MOOSE. Unfortunately peacock is not working properly on the latest macOS for me, so I cannot reproduce your issue.

Thank you for your suggestion. I will do that.

**Reply by GiudGiud** [reply]
You can add command line options in `other options` with peacock. See this page
https://mooseframework.inl.gov/application_usage/peacock.html
